src/guiapi/src/display_helper.c

```diff
@@ -52,6 +58,94 @@ void render_text_align(rect_ui16_t rc, const char *text, font_t *font, color_t c
     }
 }
 
+void scroll_text_phasing(int16_t win_id, font_t * font, txtroll_t * roll){
+    if(roll->setup == 2)
+        return;
+
+    switch (roll->phase) {
+        case ROLL_SETUP:
+            gui_timer_create_periodical(TEXT_ROLL_DELAY_MS, win_id);
+            if (roll->setup == 1)
+                roll->phase = ROLL_GO;
+            window_invalidate(win_id);
+            break;
+        case ROLL_GO:
+            if (roll->count > 0 || roll->px_cd > 0) {
+                if (roll->px_cd == 0) {
+                    roll->px_cd = font->w;
+                    roll->count--;
+                    roll->progress++;
+                }
+                roll->px_cd--;
+                window_invalidate(win_id);
+            } else {
+                roll->phase = ROLL_STOP;
+            }
+            break;
+        case ROLL_STOP:
+            gui_timers_delete_by_window_id(win_id);
```

Comment:
this call tries to delete timers and is called from a timer handler function. So far it doesn't cause any issues, but may be dangerous in the future (the caller ma keep the timer for something and it will get destroyed here)